### PR TITLE
Make Silos optional

### PR DESCRIPTION
**Pull request by iLLiCiTiT**
PR related to core's PR https://github.com/getavalon/core/pull/535.
- this is version of launcher with ability to work with or without silo
- most of changes happened a long time ago, so feel free to comment, modify or even create better PR, this works but I would say it is not clean code

### CODE REVIEW

**Comment by mottosso** [comment]
Thanks. I don't mind that it's dirty. As far as I'm concerned, Launcher is on life-support until Allzpark replaces it. @davidlatwe Are you using Launcher? If you're happy with the changes, I'm happy.

File: launcher/control.py [hunk 1]
```diff
@@ -276,27 +287,44 @@ def on_project_changed(self, index):
...
+                "type": asset["type"],
+                "icon": DEFAULTS["icon"]["asset"]
+            }) for asset in assets])
+
+        else:
+            self._model.push([dict({
+                "name": silo, "icon": DEFAULTS["icon"]["silo"], "type": "silo"
+            }) for silo in sorted(silos)])
 
+        frame = project
```
Comment (davidlatwe) [inline launcher/control.py]: Why removing `frame = self.current_frame()`, `frame["config"] = project["config"]` and use entire project document as frame ?

Is this necessary ? If not, I would like to keep it :D

**Review by davidlatwe — COMMENTED** [review]
Yeah, still using *life-support* here. 🙋‍♂ 
Just one question about this change. 👇

File: launcher/control.py [hunk 2]
```diff
@@ -187,18 +195,24 @@ def command(self, command):
...
+        frame = self.current_frame()
+
+        # If nothing is set as current frame we are at selecting projects
+        handler = self.on_project_changed
+        if frame:
+            handler = {
+                "project": self.on_asset_changed,
+                "silo": self.on_asset_changed,
+                "asset": self.on_asset_changed
+            }[frame["type"]]
```
Comment (davidlatwe) [inline launcher/control.py]: It look like `frame` in first launch did not have key `"type"`, so the above if-statement should change into
```python
if frame and frame.get("type"):
    ...
````

File: launcher/control.py [hunk 3]
```diff
@@ -248,15 +261,13 @@ def init(self):
...
                 "icon": DEFAULTS["icon"]["project"],
+                "type": project["type"],
                 "name": project["name"],
             }, **project["data"])
             for project in sorted(io.projects(), key=lambda x: x['name'])
             if project["data"].get("visible", True)  # Discard hidden projects
         ])
 
-        frame = {"environment": {}}
-        self._frames[:] = [frame]
```
Comment (davidlatwe) [inline launcher/control.py:258]: We need these two lines or `KeyError: environment` will raised later in `on_project_changed` (at line 332).

**Review by davidlatwe — CHANGES_REQUESTED** [review]
Okay, sorry for late review but I just re-visited this and have a few tests.
Once these fixed, we are good to go !

**Review by davidlatwe — APPROVED** [review]

**Comment by davidlatwe** [comment]
Thanks @iLLiCiTiT , although this PR is backward compatible (quick tested), but let's wait for getavalon/core#535 to be merged for clarity.

**Pull request closed without merging**